### DISCUSSION TITLE
CSS cache is not flushed automatically

### DISCUSSION THREAD

**Discussion by Hlavtox** [PrestaShop/PrestaShop#33400] [Ideas]
#### Describe the bug
When you had CSS compress and combine enabled in Prestashop 1.6, it used to flush the cache automatically when some of the source CSS were changed.
This no longer happens, is it done on purpose?

#### Expected behavior
I think the behavior should be the same as in TPL files, which listen for changes, if it's set so.

#### Steps to reproduce
1. Set your CCC setting like this:
![Výstřižek](https://user-images.githubusercontent.com/6097524/79452700-d91e4000-7fe8-11ea-9703-314f5af2e027.JPG)
2. Open some page in FO to cache the CSS.
3. Go to themes/themename/assets/css and change something in custom.css
You can add ` body{background-color: red !important;}` for example.
4. Go to FO, the changes are not visible.
5. Clear cache manually or delete files in themes/themename/assets/cache and you will see the change.

#### Additional information

* PrestaShop version: 1.7.6.2
* PHP version: 7.2

**Comment by florine2623** [comment]
Hi @Hlavtox ,
The cache is cleared whenever a file is modified if the configuration is the following :
![Capture d’écran 2020-04-16 à 10 47 12](https://user-images.githubusercontent.com/16019289/79435205-a6b41900-7fcf-11ea-8951-22a77890af65.png)
I've tried changing the theme.css file of the classic template. After refreshing my page, I saw the updates. It works in both 1.7.6.4 and 1.7.5.2.

**Comment by Hlavtox** [comment]
@florine2623 I am 99% sure that the settings in the screenshot are only related to TPL file changes.

1. Set your CCC setting like this:
![Výstřižek](https://user-images.githubusercontent.com/6097524/79452700-d91e4000-7fe8-11ea-9703-314f5af2e027.JPG)

2. Open some page in FO to cache the CSS.

3. Go to themes/themename/assets/css and change something in custom.css
You can add ` body{background-color: red !important;}` for example.
		
**4. Go to FO, the changes are not visible.**

5. Clear cache manually or delete files in themes/themename/assets/cache and you will see the change.

**Comment by florine2623** [comment]
Thanks for the details @Hlavtox.
I managed to reproduce the issue with PrestaShop version 1.7.6.4 and 1.7.5.2.
Indeed, when those 3 options are set in CCC, you need to clear cache manually to see css changes.
I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by sam-pires** [comment]
hola @PrestaShop/prestashop-product-team,
The behavior here needs to be investigated, does someone know what was the reason for the change between 1.6 and 1.7?

**Comment by hibatallahAouadni** [comment]
Ping @PrestaShop/product-team friendly reminder :wink:

**Comment by Hlavtox** [comment]
@hibatallahAouadni No time in 11 months :(

**Comment by MatShir** [comment]
@PrestaShop/prestashop-core-developers what do you think about this feature?

**Comment by Hlavtox** [comment]
It's not a feature, it's a bug. It worked correctly in 1.6. ;)

**Comment by matks** [comment]
It is a bit similar to https://github.com/PrestaShop/PrestaShop/issues/10091 @Hlavtox if I am not wrong?

**Comment by Hlavtox** [comment]
@matks Yeah, but that issue was not about that in the beginning I think. Keep whatever issue, but don't make it forgotten. ;)

**Comment by kpodemski** [comment]
@Hlavtox I would like to improve CCC in PrestaShop and add a QS at the end of the assets path which will be equal to `PS_CCCJS_VERSION` or `PS_CCCCSS_VERSION` - wdyt? clearing the cache would be still required but it will work better

we could also think about having an option to clear only the assets cache

**Reply by Hlavtox** [reply]
@kpodemski Not sure I understand the reason? 🤔 The name of CCCd asset change on each cache flush, it doesn't need a QS at the end. ;-)
